test/test_c_parser.py

```diff
@@ -29,6 +30,9 @@ def check_input_strings_for_grammar(
 
 def verify_ast_generation(source: str, stmt: str, tmp_path: PurePath) -> None:
     grammar = parse_string(source, GrammarParser)
+    # Check that the working directory is empty: reusing non-empty temporary
+    # directories when generating extensions can lead to segmentation faults.
+    assert not os.listdir(tmp_path)
```

Comment:
Wouldn't it be a good idea to add this assertion to `check_input_strings_for_grammar` as well?

Reply:
Good point. Actually, I am going to move this to `generate_parser_c_extension` so we don't need to duplicate this code.